content/blog/lift-and-transform-apps-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,283 @@
+kind: Secret
+metadata:
+  name: hpe-backend
+  namespace: hpe-storage
+stringData:
+  serviceName: nimble-csp-svc
+  servicePort: "8080"
+  backend: <Your Nimble array>
+  username: <Your username with privileges to the vVol folder>
+  password: <Your password>
+```
+
+A `StorageClass` needs to be specifically crafted to allow users to import volumes and snapshots from the storage array.
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: hpe-transform
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/fstype: xfs
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  description: Volume transformed from vVol by HPE CSI Driver
+  allowOverrides: importVolAsClone,importVolumeName,forceImport
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+The important details in the `StorageClass` are the `allowOverrides`. Here we put the parameters we want the application admin to override once testing has started (`importVolAsClone`) and once the database is ready for the final import (`importVolumeName` and `forceImport`).
+
+For the final production instance, we’re going to use the [Bitnami MariaDB Helm chart](https://artifacthub.io/packages/helm/bitnami/mariadb). For this chart to install cleanly and be able to bring up the database, the following considerations needs to be taken into account:
+
+* There needs to be a known `root@localhost` account with a known password in the source database.
+* The Bitnami chart does not support altering the `mountPath`. The source filesystem needs to have a symlink to `/var/lib/mysql` from `/var/lib/data`.
+
+For the initial cloning of the source database, the following `values.yaml` file is being used.
+
+```yaml
+image:
+  tag: 10.3
+auth:
+  rootPassword: my-password
+global:
+  imagePullSecrets: # Only needed if you're rate limited to Docker Hub
+  - regcred
+primary:
+  persistence:
+    enabled: true
+    storageClass: hpe-transform
+    annotations:
+      csi.hpe.com/importVolAsClone: virt-mm-db-1.vmdk
+    size: 64Gi
+    subPath: lib
+```
+
+The above parameters are quite self-explanatory, the `subPath` directive instructs the kubelet to mount the `lib` directory in the root of the volume. As a result, `/lib` in the volume will be mounted at `/bitnami/mariadb` and the `data` subdirectory which hold the MariaDB databases will be seen properly by the container binaries.
```

Comment:
Split the first sentence into two:
"The above parameters are quite self-explanatory. The subPath directive instructs..."

Add a comma prior to "which" in the last sentence, change "hold" to "holds" and another comma prior to "will" so it reads:

"... and the data sudirectory, which holds the MariaDB databases, will be seen..."